linode/helper/objects.go

```diff
@@ -20,14 +22,37 @@ func S3ConnFromResourceData(d *schema.ResourceData) *s3.S3 {
 	accessKey := d.Get("access_key").(string)
 	secretKey := d.Get("secret_key").(string)
 	cluster := d.Get("cluster").(string)
+	endpoint := d.Get("endpoint").(string)
 
-	sess := session.New(&aws.Config{
+	if endpoint == "" {
+		// fall back for legacy safety
+		endpoint = fmt.Sprintf(LinodeObjectsEndpoint, cluster)
+	}
+
+	return s3.New(session.New(&aws.Config{
 		// This region is hardcoded strictly for preflight validation purposes.
 		Region:      aws.String("us-east-1"),
 		Credentials: credentials.NewStaticCredentials(accessKey, secretKey, ""),
-		Endpoint:    aws.String(fmt.Sprintf(LinodeObjectsEndpoint, cluster)),
-	})
-	return s3.New(sess)
+		Endpoint:    aws.String(endpoint),
+	}))
+}
+
+// ComputeS3Endpoint sets endpoint in data for the s3 connection via the cluster and bucket combination
+func ComputeS3Endpoint(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
+	if d.Get("endpoint").(string) != "" {
+		return nil
+	}
```

Comment:
We probably can considering removing this check. Usually terraform needs to refresh the data every time when it's read by user.

Reply:
@lgarber-akamai and i chatted and there are some more fixes for how this is being done and ill have an updated PR soon. agree the check can be removed... this is ack'ed and resolving the convo for now and my new changes will come today